[Tracking] Liana business installer

pythcoiner opened the pull request:
- [x] **2. Back**
  - [x] 2.1 Auth Client
  - [x] 2.2 Installer Trait Integration
  - [x] 2.3 WSS Protocol Extraction
- [x] **3.1 WS Manager Flow**
- [ ] **1. Front**
  - [x] 1.1 Wallet Selection View
  - [x] 1.2 Edit Wallet Template Subflow (needed for completion of WS Manager flow)
  - [x] 1.3 Add Key Information Subflow
  - [x] 1.4 Filter/Search Bar (WS Manager Only)
  - [x] 1.5 Better Keyboard Navigation in Login
  - [ ] 1.6 Load Wallet Subflow
  - [x] 1.7 Logout Feature
- [x] **3.2 Owner Flow**
- [x] **3.3 Participant Flow**
- [x] **4. Local Storage**

# Comments addressed:

## Bugs/Errors
- [x] Fix panic when running in debug mode (`cargo run --bin liana-business`)

## Features/Questions to Address
- [x] Implement "lock" of template by WS manager 

# Comments NOT addressed:
- [!] Fix application panic when trying to load finalized wallets
- [!] Add "Add wallet" button (or clarify if wallet creation is server-side only) 
- [!] Consider supporting multiple panels (for WS Manager and testing scenarios)
- [!] Consider adding organization creation from GUI (potentially for future release)

## UI/Wording Changes

### Wallet Status Labels (Wallet List View)
- [x] Change "Validated" status label
- [x] Add status badge for finalized wallets

### Manage Key Modal (Key editing popup)
- [x] Rename "Alias" field label
- [x] Rename "Description" field label
- [x] Rename "Email" field label
- [x] Clarify "Key type" dropdown

### Key Information Screen (Screen showing keys that need xpubs)
- [x] Change "Missing" status indicator
- [x] Update introductory/instructional text
- [x] Rename screen title and header

### Header/Breadcrumb Naming
- [x] Update Organizations list header
- [x] Update Wallets list header/breadcrumb


# To do:
- [x] Cache bitbox pairing code
- [x] Test Jade
- [x] Test specter
- [x] For WsManager role, in org list, the wallet count in in correct, not visible wallets are counted
- [x] create an icon for liana-business (svg?)
- [x] in login page if wrong email entered, send token is greyd out, user stuck (likely dummy server)
- [x] owner role, Manage Key button is displayed after wallet approved
- [x] add liana-business to CI
- [x] rename roles in the UI
- [ ] settings
- [x] fetch all users for all `last_editor`
- [ ] Support Mainnet & signet
- [x] Installer => register descriptor as last step
- [x] pass runtime to async-hwi::service
- [ ] if split the window and move tab at login(mail) step, seems we lost the notification stream, its fine if we start from a new tab from the splitted pane

pythcoiner committed:
repo: move dependency version to workspace

pythcoiner committed:
liana-business: wip

pythcoiner committed:
update wss protocol & Backend trait

pythcoiner committed:
wss_client wip

pythcoiner committed:
crossbeam to avoid spinloop

pythcoiner committed:
wss-client: reorg

pythcoiner committed:
client: check requests id

pythcoiner committed:
client: unit tests

pythcoiner committed:
client: intefration tests

pythcoiner committed:
client: sync implem / test / spec

pythcoiner committed:
use DummyServer as backend for dev environment

pythcoiner committed:
cleanup

pythcoiner committed:
roadmap

pythcoiner committed:
implement auth

pythcoiner committed:
implement Installer for BusinessInstaller

pythcoiner committed:
extract wss protocol to liana-connect

pythcoiner committed:
docs & cleanup

pythcoiner committed:
roadmap

pythcoiner committed:
cleanup liana-connect

pythcoiner committed:
wallet selection view

pythcoiner committed:
improve keyboard navigation in login flow

pythcoiner committed:
search/filter bar for wallets & orgs

pythcoiner committed:
implem logout

pythcoiner committed:
rework template visualization and editing

pythcoiner committed:
refactor manage keys view

pythcoiner committed:
connect ws after login success

pythcoiner committed:
implem dummy remote server

pythcoiner committed:
handle conflict on updates from server

pythcoiner committed:
remove embeded dummy server and use remote one

pythcoiner committed:
add breadcrumb navigation in layout header

pythcoiner committed:
implem caching auth token

pythcoiner committed:
cleanup

pythcoiner committed:
xpubs

pythcoiner committed:
notif listener

pythcoiner committed:
use async-hwi::service

pythcoiner committed:
UI cleanup

pythcoiner committed:
debug & fix dummy server

nondiremanuel commented:
I get the following error when running in debug mode (`cargo run --bin liana-business`):

```
warning: `business-installer` (lib) generated 7 warnings
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.78s
     Running `target/debug/liana-business`
  2025-12-31T08:49:33.678665Z  WARN sctk_adwaita::buttons:172: Ignoring unknown button type: 

  2025-12-31T08:49:34.438241Z  WARN sctk_adwaita::buttons:172: Ignoring unknown button type: 

  2025-12-31T08:49:58.944440Z DEBUG business_installer::state::update:36: received message, LoginUpdateEmail("use@example.com")

  2025-12-31T08:49:59.070784Z DEBUG business_installer::state::update:36: received message, LoginUpdateEmail("us@example.com")

  2025-12-31T08:49:59.226830Z DEBUG business_installer::state::update:36: received message, LoginUpdateEmail("usw@example.com")

  2025-12-31T08:49:59.419818Z DEBUG business_installer::state::update:36: received message, LoginUpdateEmail("us@example.com")

  2025-12-31T08:49:59.574038Z DEBUG business_installer::state::update:36: received message, LoginUpdateEmail("u@example.com")

  2025-12-31T08:49:59.681294Z DEBUG business_installer::state::update:36: received message, LoginUpdateEmail("uw@example.com")

  2025-12-31T08:49:59.796058Z DEBUG business_installer::state::update:36: received message, LoginUpdateEmail("u@example.com")

  2025-12-31T08:49:59.932966Z DEBUG business_installer::state::update:36: received message, LoginUpdateEmail("@example.com")

  2025-12-31T08:50:00.168172Z DEBUG business_installer::state::update:36: received message, LoginUpdateEmail("w@example.com")

  2025-12-31T08:50:00.378196Z DEBUG business_installer::state::update:36: received message, LoginUpdateEmail("ws@example.com")

  2025-12-31T08:50:02.201836Z DEBUG business_installer::state::update:36: received message, LoginSendToken

  2025-12-31T08:50:02.201971Z DEBUG business_installer::client:889: auth_request: starting for email=ws@example.com

  2025-12-31T08:50:02.205825Z DEBUG business_installer::client:895: auth_request: fetching service config for network=Bitcoin

thread '<unnamed>' panicked at /home/nondiremanuel/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tokio-1.43.0/src/runtime/blocking/shutdown.rs:51:21:
Cannot drop a runtime in a context where blocking is not allowed. This happens when a runtime is dropped from within an asynchronous context.
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
  2025-12-31T08:50:02.207144Z ERROR reqwest::blocking::client:1084: Failed to communicate successful startup: Ok(())

```

pythcoiner committed:
record last_edited & last_editor for user/org/template/key

pythcoiner committed:
track xpub source

pythcoiner committed:
require WsManager to "lock" the template prior the Owner can validate it

pythcoiner committed:
enforce Participant roles cannot access Draft/Locked wallets

pythcoiner committed:
cleanup

pythcoiner committed:
debug

pythcoiner committed:
clear client cache on logout

pythcoiner committed:
add key description to xpub card

pythcoiner committed:
add last_edited/last_editor fields to SpendingPathJson

pythcoiner committed:
fix network to Signet in liana-business

pythcoiner committed:
add serde default for SpendingPathJson fields

pythcoiner committed:
add DEBUG logging for xpub source information

pythcoiner commented:
> Application panics when trying to load finalized wallets:

this part is not yet implemented, the actual binary is just a wrapper around the installer, it does not integrate the wallet

> 8. Isn’t there the possibility for multiple panels (especially for WS Manager and testing)? 

idem, the installer itself will be in a "normal" panel as it is in Liana

> I don’t see a “Add wallet” button. Is this something we count on managing on server side?

> Should we be able to add an organization from the GUI too? (maybe for the future)

for both this point, my understanding in @edouardparis wanted to keep the admin part on the web interface

pythcoiner committed:
change Validated status badge to 'Set keys' with warning style

pythcoiner committed:
add Active status badge for finalized wallets

pythcoiner committed:
rename key modal field labels for clarity

pythcoiner committed:
add tooltip to Key Type field with descriptions

pythcoiner committed:
change Missing badge text to Not Set

pythcoiner committed:
rename Set Keys screen title and breadcrumb

pythcoiner committed:
update Set Keys intro text with hardware device recommendation

pythcoiner committed:
pluralize Organization and Wallet breadcrumbs

pythcoiner committed:
do to spin loop in ws thread

pythcoiner committed:
fix async-hwi service

pythcoiner committed:
async-rust: how to make very simple things very complex

pythcoiner committed:
allow owner to manage keys if wallet not locked

pythcoiner commented:
> When I close and reopen the app, logging back in with a previously connected account, I (sometimes) don't see the wallet list anymore:

I've also got that, but I think it's something on the dummy server, and we will drop the code after the prod server is implemented...

pythcoiner committed:
update roadmap

pythcoiner committed:
update key type tooltips for Co-signer and Safety Net

pythcoiner committed:
change Populated label to Set in xpub view

pythcoiner committed:
add waiting state in Set Keys view when all keys are set

pythcoiner committed:
Lock template => Send for approval

pythcoiner committed:
Validate Template => Approve Template

pythcoiner commented:
> In general, I re-thought about the terminology we use for the "owner" and I think it could be misleading. I propose to use "Admin" everywhere for better understanding. Maybe we should also be more specific with "Participant" by calling him "Signer", or alternatively staying general but with a better term, such as "Member".

I dont think thos terms are used in the UI?

pythcoiner committed:
update roadmap for liana-business

pythcoiner committed:
move business-installer inside liana-business

pythcoiner committed:
make settings generic

pythcoiner committed:
prepare generic settings for liana-business

pythcoiner committed:
installer: add skip_launcher() to Installer trait for initial state control

Adds a new method to the Installer trait that allows implementations to
specify whether to skip the Launcher and start directly with the Installer.
Default is false (show Launcher). This enables liana-business to start
directly with authentication in the installer.

nondiremanuel commented:
> > In general, I re-thought about the terminology we use for the "owner" and I think it could be misleading. I propose to use "Admin" everywhere for better understanding. Maybe we should also be more specific with "Participant" by calling him "Signer", or alternatively staying general but with a better term, such as "Member".
> 
> I dont think thos terms are used in the UI?

What UI? I didn't understand the question sorry.
 I'm proposing to use "Admin" and "Signer" in the UI too.

pythcoiner committed:
integrate GUI into liana-business

pythcoiner committed:
cleanup liana-business docs

pythcoiner committed:
auto refresh token

pythcoiner committed:
installer => app flow

pythcoiner committed:
move liana-connect into liana-business

pythcoiner committed:
bump rust version in CI & clippyfy

pythcoiner committed:
fix(org_select): apply hide_finalized filter to wallet count

pythcoiner committed:
fix(login): reset processing flag on generic backend errors

pythcoiner committed:
fix(template_builder): hide Manage Keys button for Owner after approval

pythcoiner committed:
roadmap

pythcoiner committed:
liana-gui: make parse_args() composable

pythcoiner committed:
liana-gui: move parse_args() to lib

pythcoiner committed:
do not reimplement args parsing

pythcoiner committed:
liana-gui: refactor main

pythcoiner committed:
liana-gui: move main subfunctions to lib

pythcoiner committed:
liana-business: rework main

pythcoiner committed:
clippyfy

pythcoiner committed:
liana-business: use validation methods

pythcoiner committed:
liana-business: cleanup
